### PR TITLE
Makes all files owned by root

### PR DESCRIPTION
**Pull request by svavassori**
When creating a .deb package, all files should be owned by root as implemented in dh_fixperms, unless there is a specific reason to use a different user (e.g. `preinst` script creates that user).

### CODE REVIEW

**Comment by jiro4989** [comment]
Thank you.
This is a breaking changes, so please give me some time to think about the change policy.

for example:

1. Create a `v3.0.0` tag and make a major version upgrade
2. or, Add `root_owner_group` parameter (default false) to action.yaml. This allows the user to enable the option at any time (not breaking changes)

**Comment by svavassori** [comment]
Sure, no hurry.

I was also having a second though about it because of that and in the case you would go with the mayor version upgrade, I would still add a `keep_ownership` parameter (default `false`) so the default case would set ownership to `root` (since it's the most common one), but leave the option to the user to define a different ownership.

If it's OK for you, once you have taken a decision, I'll update the PR accordingly.

**Comment by jiro4989** [comment]
I decided to upgrade the major version.

Lets do with this:

> I was also having a second though about it because of that and in the case you would go with the mayor version upgrade, I would still add a keep_ownership parameter (default false) so the default case would set ownership to root (since it's the most common one), but leave the option to the user to define a different ownership.

Could you update this PR?

**Comment by svavassori** [comment]
Hello,

I have updated the PR with 3 commits:

1. it adds the `keep_ownership` parameter in order to select which ownership to build the package with
2. it fixes a missing valid compression option (`none`) for `dpkg-deb` while sanitizing the input parameter `compress_type`
3. since we are already making a breaking change, I thought it would make sense to take advantage of it and set the default `compress_type` to match the default one of `dpkg-deb`

Since the third commit is something we didn't talk about (I just noticed it while fixing the `none` one), I left the last commit.
Please, let me know if it's OK for you to include it in this PR.


About the syntax for parameter expansion, I used this one `${parameter:+word}`: _If parameter is null or unset, nothing is substituted, otherwise the expansion of word is substituted._

https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html

**Comment by jiro4989** [comment]
Thank you.
I would like to discuss this:

> since we are already making a breaking change, I thought it would make sense to take advantage of it and set the default compress_type to match the default one of dpkg-deb

There is a reason why `compress_type` defaults to `gzip`.
See: https://github.com/jiro4989/build-deb-action/issues/40

The default value of `compress_type` changed from `gzip` to `zstd` when i had upgraded the Docker base image of `build-deb-action`. As a result, some users' environments could not handle `zstd` and errors occurred.

`zstd` seems to be a fast compression algorithm, but I didn't see the benefit of breaking changes. Therefore, I think that it is not necessary to match the default behavior of `dpkg-deb`. I don't think `zstd` is popular yet, so I think it's too early to change it.

But I'm wondering if I should change it. Please let me your opinion.

**Comment by jiro4989** [comment]
Oh, I have updated the `master` branch, please `git rebase`.

**Comment by svavassori** [comment]
Hello @jiro4989,

I wasn't aware about issue #40 and it brings up an interesting topic (beside making me realize how time flies :-| )

I've looked at `dpkg-deb` compress-types across some Ubuntu releases and this is the compatibility matrix, the ❗ sign means it has been deprecated while ✔️ means it is the default compress-type.


 Ubuntu version | gzip | xz | bzip2 | lzma | none | zstd
-- | -- | -- | -- | -- | -- | --
14.04 trusty (LTS) | ✅ | ✔️  | ✅ | ✅ | ✅ |
16.04 xenial (LTS) | ✅ | ✔️ | ✅❗| ✅❗ | ✅ |  
18.04 bionic (LTS) | ✅ | ✔️ |   |   | ✅ |  
20.04 focal (LTS) | ✅ | ✔️ |   |   | ✅ |
20.10 groovy | ✅ | ✔️ |   |   | ✅ |✅
21.04 hirsute | ✅ | ✔️ |   |   | ✅ |✅
21.10 impish | ✅ | ✅ |   |   | ✅ | ✔️
22.04 jammy (LTS) | ✅ | ✅ |   |   | ✅ | ✔️
22.10 kinetic | ✅ | ✅ |   |   | ✅ | ✔️
23.04 lunar | ✅ | ✅ |   |   | ✅ | ✔️


Since I'm using Debian testing I wouldn't have noticed it, but current Debian stable (11 -  bullseye) doesn't have support for it, so I think making `zstd` as the new default of the action is too early.

The problem is that whatever decision we take to select the default value has pros and cons:
1. `gzip` it is compatible with whatever existing system but it has inferior compression ratio compared with `xz` or `zstd`. Also it isn't the most used to build `.deb` packages since more than 10 years, so it's kind of legacy and it _may_ be deprecated in the future in favor of `zsdt` (like `xz` made `bzip2` obsolete).
2. `xz` it is the current most widely used and it will work (practically) on all machines, it has a better compression ration but it is somewhat slower than `gzip`
3. `zstd` probably will be the "new" general default, it's faster in compressing and decompressing than `xz` or `gzip` and has a compression ratio similar or better than `gzip`
4. `none` could also be a reasonable default: it's fast, it won't obsolete and it will work on every machine, but doesn't compress
5. if we don't pick any value and use the "dpkg-deb defaul" we should take care to check if it has changed when upgrading the docker image and bumping the major version accordingly.

The last point brings the same experience (and problems) of using directly `dpkg-deb` from the command line, so it's the most transparent and it fits the aim of GitHub action to create `.deb`, but I agree it isn't really nice to bump up a major number just because it changed the underling default behavior.

Since this is a GitHub action to _package_ (same debate applies to [RPM](https://fedoraproject.org/wiki/Changes/Switch_RPMs_to_zstd_compression)), I think it would be worth to make the user aware that the `compress_type` is a sensible parameter that has impact on how the `.deb` is created and consumed (storage, bandwidth, memory, CPU time, compatibility, etc) and there is not a global optimum because it depends on the use of this `.deb` (e.g. for pure archiving or downloading through a slow connection, size is king, but for cloud ops, probably (de)compression time is a paramount).


So, to sum up, I'm fine with all these three options:

1. keep the current default `gzip`
2. set the default to `none`
3. do not have a default and change the `compress_type` to a required parameter.


Let me know which one do you prefer and I'll update the PR accordingly.

**Comment by jiro4989** [comment]
Thank you for your detailed description. I will check and reply when my job is done.

**Comment by jiro4989** [comment]
> keep the current default gzip

Lets do this.

I was surprised that the previous default `compress_type` was `xz`, not `gzip`.
(My last change was a breaking change😓)
But I don't think it will be deprecated for some time.
If `gzip` is deprecated, then `zstd` will also be popular.
Then I think again about the default value of `compress_type`.

**Comment by jiro4989** [comment]
Great!
All CI tests passed.
Thank you for your contribution.

LGTM👍

LGTM
